Test failures with assert_roundtrip_tree (from asdf)

There is a bunch of new failures in the tests (example : https://travis-ci.org/astropy/astropy/jobs/330186208), which appears to be due to a change in the `assert_roundtrip_tree` imported from asdf. This is because of https://github.com/spacetelescope/asdf/commit/dd4f32f05c1d91417d0887ba6c1acaadc2a519e6 - ping @Cadair @drdavella 

```
    def test_table(tmpdir):
        data_rows = [(1, 2.0, 'x'),
                     (4, 5.0, 'y'),
                     (5, 8.2, 'z')]
        t = table.Table(rows=data_rows, names=('a', 'b', 'c'),
                        dtype=('i4', 'f8', 'S1'))
        t.columns['a'].description = 'RA'
        t.columns['a'].unit = 'degree'
        t.columns['a'].meta = {'foo': 'bar'}
        t.columns['c'].description = 'Some description of some sort'
    
        def asdf_check(ff):
            assert len(ff.blocks) == 3
    
>       helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check)
E       TypeError: assert_roundtrip_tree() takes 2 positional arguments but 3 were given
```

--- a/astropy/io/misc/asdf/tags/table/tests/test_table.py
+++ b/astropy/io/misc/asdf/tags/table/tests/test_table.py
@@ -20,10 +20,10 @@ def test_table(tmpdir):
     t.columns['a'].meta = {'foo': 'bar'}
     t.columns['c'].description = 'Some description of some sort'
 
-    def asdf_check(ff):
+    def check(ff):
         assert len(ff.blocks) == 3
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check)
+    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check)
 
 
 def test_array_columns(tmpdir):
@@ -37,10 +37,10 @@ def test_array_columns(tmpdir):
     t = table.Table(a, copy=False)
     assert t.columns['a'].shape == (3, 2, 2)
 
-    def asdf_check(ff):
+    def check(ff):
         assert len(ff.blocks) == 1
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check)
+    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check)
 
 
 def test_structured_array_columns(tmpdir):
@@ -54,10 +54,10 @@ def test_structured_array_columns(tmpdir):
 
     t = table.Table(a, copy=False)
 
-    def asdf_check(ff):
+    def check(ff):
         assert len(ff.blocks) == 1
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check)
+    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check)
 
 
 def test_table_row_order(tmpdir):
@@ -74,10 +74,10 @@ def test_table_row_order(tmpdir):
     t.columns['a'].meta = {'foo': 'bar'}
     t.columns['c'].description = 'Some description of some sort'
 
-    def asdf_check(ff):
+    def check(ff):
         assert len(ff.blocks) == 1
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check)
+    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check)
 
 
 def test_table_inline(tmpdir):
@@ -91,10 +91,10 @@ def test_table_inline(tmpdir):
     t.columns['a'].meta = {'foo': 'bar'}
     t.columns['c'].description = 'Some description of some sort'
 
-    def asdf_check(ff):
+    def check(ff):
         assert len(list(ff.blocks.internal_blocks)) == 0
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check,
+    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check,
                                   write_options={'auto_inline': 64})
 
 
@@ -131,7 +131,7 @@ def test_masked_table(tmpdir):
     t.columns['a'].mask = [True, False, True]
     t.columns['c'].description = 'Some description of some sort'
 
-    def asdf_check(ff):
+    def check(ff):
         assert len(ff.blocks) == 4
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check)
+    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check)

--- a/astropy/io/misc/asdf/tags/transform/tests/test_transform.py
+++ b/astropy/io/misc/asdf/tags/transform/tests/test_transform.py
@@ -47,7 +47,7 @@ def test_inverse_transforms(tmpdir):
     def check(ff):
         assert ff.tree['rotation'].inverse.angle == 45
 
-    helpers.assert_roundtrip_tree(tree, tmpdir, check)
+    helpers.assert_roundtrip_tree(tree, tmpdir, asdf_check_func=check)
 
 
 @pytest.mark.parametrize(('model'), test_models)
@@ -61,7 +61,7 @@ def test_name(tmpdir):
         assert ff.tree['rot'].name == 'foo'
 
     tree = {'rot': astmodels.Rotation2D(23, name='foo')}
-    helpers.assert_roundtrip_tree(tree, tmpdir, check)
+    helpers.assert_roundtrip_tree(tree, tmpdir, asdf_check_func=check)
 
 
 def test_zenithal_with_arguments(tmpdir):
@@ -83,7 +83,7 @@ def test_naming_of_compound_model(tmpdir):
     tree = {
         'model': model
     }
-    helpers.assert_roundtrip_tree(tree, tmpdir, asdf_check)
+    helpers.assert_roundtrip_tree(tree, tmpdir, asdf_check_func=asdf_check)
 
 
 def test_generic_projections(tmpdir):

